fvm_actor_utils/src/util.rs

```diff
@@ -0,0 +1,201 @@
+    fn initialize_account(&self, address: &Address) -> MessagingResult<ActorID>;
+
+    /// Resolves an address to an ID address, sending a message to initialize an account there if
+    /// it doesn't exist
+    ///
+    /// If the account cannot be created, this function returns MessagingError::AddressNotInitialized
+    fn resolve_or_init(&self, address: &Address) -> MessagingResult<ActorID> {
+        let id = match self.resolve_id(address) {
+            Ok(addr) => addr,
+            Err(MessagingError::AddressNotResolved(_e)) => self.initialize_account(address)?,
+            Err(e) => return Err(e),
+        };
+        Ok(id)
+    }
+
+    /// Attempts to compare two addresses, seeing if they would resolve to the same Actor without
+    /// actually instantiating accounts for them
+    ///
+    /// If a and b are of the same type, simply do an equality check. Otherwise, attempt to resolve
+    /// to an ActorID and compare
+    fn same_address(&self, address_a: &Address, address_b: &Address) -> bool {
+        let protocol_a = address_a.protocol();
+        let protocol_b = address_b.protocol();
+        if protocol_a == protocol_b {
+            address_a == address_b
+        } else {
+            // attempt to resolve both to ActorID
+            let id_a = match self.resolve_id(address_a) {
+                Ok(id) => id,
+                Err(_) => return false,
+            };
+            let id_b = match self.resolve_id(address_b) {
+                Ok(id) => id,
+                Err(_) => return false,
+            };
+            id_a == id_b
+        }
+    }
+
+    /// Get the root cid of the actor's state
+    fn root_cid(&self) -> ActorResult<Cid>;
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct FvmActor {
+    fvm_runtime: runtime::FvmRuntime,
+}
+
+impl Actor for FvmActor {
+    fn actor_id(&self) -> ActorID {
+        self.fvm_runtime.receiver()
+    }
+
+    fn send(
+        &self,
+        to: &Address,
+        method: MethodNum,
+        params: RawBytes,
+        value: TokenAmount,
+    ) -> MessagingResult<Receipt> {
+        Ok(self.fvm_runtime.send(to, method, params, value)?)
+    }
+
+    fn resolve_id(&self, address: &Address) -> MessagingResult<ActorID> {
+        self.fvm_runtime
+            .resolve_address(address)
+            .ok_or(MessagingError::AddressNotResolved(*address))
+    }
+
+    fn initialize_account(&self, address: &Address) -> MessagingResult<ActorID> {
+        if let Err(e) = send::send(address, METHOD_SEND, Default::default(), TokenAmount::zero()) {
+            return Err(e.into());
+        }
+
+        self.resolve_id(address)
+    }
+
+    fn root_cid(&self) -> ActorResult<Cid> {
+        Ok(self.fvm_runtime.root()?)
+    }
+}
+
+impl Blockstore for FvmActor {
+    fn get(&self, cid: &Cid) -> anyhow::Result<Option<Vec<u8>>> {
+        // If this fails, the _CID_ is invalid. I.e., we have a bug.
+        fvm_sdk::ipld::get(cid)
```

Comment:
I've made the runtimes themselves impl Blockstore as the ipld::get and ipld::set operations will be trickier to implement and will require a new FakeBlockstore impl rather than relying on MemoryBlockstore. 

Alternatively the Runtime trait could have a method that returns a reference to a Blockstore